### PR TITLE
add esc temperatur on cli status

### PR DESCRIPTION
**Pull request by Christoph-Hofmann**
Show ESC temperatures from each esc.
So you can easily see that telemetry from all escs is working.

### CODE REVIEW

**Review by DzikuVx — APPROVED** [review]

File: src/main/fc/cli.c [hunk 1]
```diff
@@ -3193,6 +3196,18 @@ static void cliStatus(char *cmdline)
         hardwareSensorStatusNames[getHwGPSStatus()]
     );
 
+#ifdef USE_ESC_SENSOR
+    uint8_t motorCount = getMotorCount();
```
Comment (JulioCesarMatias) [inline src/main/fc/cli.c:3200]: There is a check missing to only print this in the case that the user has configured the ESC sensors - no point printing it when there is no data to print.

**Comment by JulioCesarMatias** [comment]
What need is there to check the esc temperature in the CLI where the motors won't even be spinning and the temp will just be the base idle value for the esc?

**Comment by JulioCesarMatias** [comment]
The `status` command is used for users to provide status information when helping developers to debug a problem - this information is unrelated to this task, and will only add confusion, so even if the CLI is the right place for your use case, `status` isn't the right command.

**Comment by JulioCesarMatias** [comment]
In my opinion, this should be moved from the CLI, to somewhere in the iNav-Configurator.

**Comment by Christoph-Hofmann** [comment]
That the easy way to see, if all esc's are sending teletry values.
In the OSD you see only the highest temperature.
So it's for debugging purpose.

**Comment by JulioCesarMatias** [comment]
> That the easy way to see, if all esc's are sending teletry values.
> In the OSD you see only the highest temperature.
> So it's for debugging purpose.

hello, sorry for the mistake, i didn't realize it was through telemetry. Disregard my comments

File: src/main/fc/cli.c [hunk 2]
```diff
@@ -3193,6 +3196,18 @@ static void cliStatus(char *cmdline)
...
+    cliPrintLinef("ESC Temperature(s): Motor Count=%d",motorCount);
```
Comment (DzikuVx) [inline src/main/fc/cli.c:3201]: @Christoph-Hofmann this is a very cool addition. I always struggle to verify if ESC telemetry works.
Let's however, improve it and display this line onli if 
`if (STATE(ESC_SENSOR_ENABLED) && getMotorCount() > 0) {`

**Review by DzikuVx — CHANGES_REQUESTED** [review]

**Comment by avsaase** [comment]
How about just showing `ESC TELEM = OK` when telemetry is received. I agree with @JulioCesarMatias that the actual temperature values aren't that interesting on the bench. This could also be added to the configurator with the other sensors.

**Comment by rmaia3d** [comment]
> How about just showing `ESC TELEM = OK` when telemetry is received. I agree with @JulioCesarMatias that the actual temperature values aren't that interesting on the bench. This could also be added to the configurator with the other sensors.

Nooo, they are interesting and important on the bench as well!! It's a good and quick way to determine if an ESC is going bad (or there's a short or anything else wrong with the connections) before the magic smoke is let out...

**Comment by JulioCesarMatias** [comment]
> The `status` command is used for users to provide status information when helping developers to debug a problem - this information is unrelated to this task, and will only add confusion, so even if the CLI is the right place for your use case, `status` isn't the right command.

.

**Comment by JulioCesarMatias** [comment]
> In my opinion, this should be moved from the CLI, to somewhere in the iNav-Configurator.

.

**Comment by DzikuVx** [comment]
@JulioCesarMatias if you could update the PR with requested changes, I'd like to merge it to `master` quite soon

**Comment by DzikuVx** [comment]
Replaced by #7200
Thanks a lot to all the participants

**Pull request closed without merging**